sentinel-extension/sentinel-parameter-flow-control/src/main/java/com/alibaba/csp/sentinel/slots/statistic/cache/ConcurrentLinkedHashMapWrapper.java

```diff
@@ -52,45 +51,54 @@ public ConcurrentLinkedHashMapWrapper(ConcurrentLinkedHashMap<T, R> map) {
 
     @Override
     public boolean containsKey(T key) {
-        return map.containsKey(key);
+        return map.asMap().containsKey(key);
     }
 
     @Override
     public R get(T key) {
-        return map.get(key);
+        return map.getIfPresent(key);
     }
 
     @Override
     public R remove(T key) {
-        return map.remove(key);
+        return map.asMap().remove(key);
     }
 
     @Override
     public R put(T key, R value) {
-        return map.put(key, value);
+        return map.asMap().put(key, value);
     }
 
     @Override
     public R putIfAbsent(T key, R value) {
-        return map.putIfAbsent(key, value);
+        return map.asMap().putIfAbsent(key, value);
     }
 
     @Override
     public long size() {
-        return map.weightedSize();
+        return map.asMap().keySet().size();
     }
 
     @Override
     public void clear() {
-        map.clear();
+        map.invalidateAll();
+    }
+
+    @Override
+    public Set<T> keySet() {
+        return map.asMap().keySet();
     }
 
     @Override
     public Set<T> keySet(boolean ascending) {
         if (ascending) {
-            return map.ascendingKeySet();
+            return map.asMap().keySet().stream().sorted()
```

Comment:
fyi, this is supported via the extension api
https://github.com/ben-manes/caffeine/wiki/ConcurrentLinkedHashMap#snapshot-views